dotCMS/src/main/java/com/dotcms/graphql/business/PageAPIGraphQLTypesProvider.java

```diff
@@ -0,0 +1,60 @@
+package com.dotcms.graphql.business;
+
+import static graphql.Scalars.GraphQLBoolean;
+import static graphql.Scalars.GraphQLString;
+
+import com.dotcms.graphql.InterfaceType;
+import com.dotcms.graphql.util.TypeUtil;
+import com.dotcms.graphql.util.TypeUtil.TypeFetcher;
+import graphql.schema.GraphQLOutputType;
+import graphql.schema.GraphQLType;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Singleton class that provides all the {@link GraphQLType}s needed for the Page API
+ */
+
+public enum PageAPIGraphQLTypesProvider implements GraphQLTypesProvider {
+
+    INSTANCE;
```

Comment:
what is that for?

Reply:
This is a singleton implementation using ENUM, as stated in the Book Effective Java from Java creator Joshua Botch. It is the most recommended way of creating a singleton. You can see it here in Method 3. https://medium.com/codelog/3-methods-to-create-singleton-with-java-effective-java-learning-notes-12ee8cd27ccd